File: recette/src/main/java/com/example/recette/bean/Recette.java
package com.example.recette.bean;

import jakarta.persistence.*;
import lombok.Data;
import java.util.Date;
import java.util.List;

@Entity
@Data
public class Recette {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String ref;
    private String nom;
    private String description;
    private int duree;
    private Date date_publication;
    private String userRef;
    private String image;

    @OneToMany
    List<Ingredient> ingredients;
    @OneToMany
    List<Etape> etapes;

    @ManyToOne
    TypeRecette typeRecette;

}

File: front-end/src/app/models/recette.ts
import { Etape } from "./etape";
import { Ingredient } from "./ingredient";
import { Type } from "./type";

export class Recette {
  id: number;
  nom: string;
  ref: string;
  description: string;
  duree: number;
  datePublication: Date;
  ingredients?: Ingredient[];
  etapes?: Etape[];
  image: string;
  userRef: string;
  typeRecette: Type;

  constructor(
    id: number,
    nom: string,
    ref: string,
    description: string,
    duree: number,
    image: string,
    datePublication: Date,
    userRef: string,
    typeRecette: Type,
    ingredients?: Ingredient[],
    etapes?: Etape[]
  ) {
    this.id = id;
    this.ref = ref;
    this.nom = nom;
    this.image = image;
    this.description = description;
    this.duree = duree;
    this.datePublication = datePublication;
    this.userRef = userRef;
    this.typeRecette = typeRecette;
    this.ingredients = ingredients;
    this.etapes = etapes;
  }
}
